parse.lisp

```diff
@@ -143,7 +146,11 @@
     (keyword  (setf *command-line-options*
 		    (list* action value *command-line-options*)))
     (symbol   (set action value))
-    (function (funcall action value))))
+    ;; Function actions are saved as (keyword . function).
+    (cons (setf *command-line-options*
+                ;; Add the result of calling action to `*command-line-options*'.
+                (list* (car action) (funcall (cdr action) value)
+                       *command-line-options*)))))
```

Comment:
From looking at the quicklisp clients:

    CL-USER> (quicklisp-client:who-depends-on "command-line-arguments")
    ("hemlock.base" "hu.dwim.util.test" "hu.dwim.util/command-line" "hu.dwim.util/production" "pgloader" "projectured.executable" "query-fs")

Only hemlock (according to rgrep) defines an action on an option (their "backend/backend-type" options).  Passing these options through looks like it would cause them errors for undefined keyword arguments in two functions.  I could suggest a patch to hemlock or hide this behavior behind a configuration variable named `*keep-action-results*` (or somesuch) and default to the old behavior (although I do think the new behavior is a better default).

Reply:
If I merge this PR, will you send a PR to hemlock so it works with your new convention?